src/storage/createRestStorage.ts

```diff
@@ -0,0 +1,283 @@
+	return JSON.parse(response);
+}
+
+const instanceStateMap = new WeakMap<Storage<{}, {}>, RestState<{}>>();
+
+const JsonContentType = 'application/json';
+const JsonPatchContentType =  'application/json-patch+json';
+const MergePatchContentType = 'application/merge-patch+json';
+
+/*
+  Return the patch serializer for the first supported content type in 'Accept-Patch' header or user option.
+ */
+function getPatchSerializer(acceptPatch: string | false): (patch: Patch<{}, {}>) => string {
+	if (!acceptPatch || acceptPatch.trim().length === 0) {
+		return null;
+	}
+
+	for (let contentType of acceptPatch.split(',')) {
+		if (JsonPatchContentType === contentType.trim()) {
+			return (patch: Patch<{}, {}>) => patch.toString();
+		}
+		else if (MergePatchContentType === contentType.trim()) {
+			// TODO: dependency - merge patch is not implemented yet
+			// return (patch: Patch<{}, {}>) => patch.toMergeString();
+		}
+	}
+	return null;
+}
+
+const createRestStorage: RestStorageFactory = createBaseStorage.mixin(compose({
+
+	createId(this: Storage<{}, {}>): Promise<string> {
+		const state = instanceStateMap.get(this);
+		if (state.createIdTarget) {
+			return <Promise<any>> request.get(state.createIdTarget).then(function(response) {
+				return String(response.data);
+			});
+		}
+		else {
+			return Promise.reject(new Error('createId is not supported in a rest store.'));
+		}
+	},
+
+	fetch<V>(this: Storage<{}, {}>, query?: Query<{}, V>): Promise<V[]> {
+		const state = instanceStateMap.get(this);
+		let uri = state.target;
+		if (query) {
+			uri += '?' + query.toString(state.querySerializer);
+		}
+		return <Promise<any>> request.get(uri, {
+			headers: {
+				'Accept': state.contentType
+			}
+		}).then(function(response) {
+			return state.responseDeserializer(String(response.data));
+		});
+	},
+
+	get(this: Storage<{}, {}>, ids: string[]): Promise<{}[]> {
+		if (!ids || ids.length === 0) {
+			return Promise.reject(new Error('ids can\'t be empty.'));
+		}
+		const state = instanceStateMap.get(this);
+		let uri = state.target;
+
+		if (ids.length === 1) {
+			uri += '/' + ids[0];
+		}
+		else {
+			uri += '?' + state.idSerializer(ids);
+		}
+
+		return <Promise<any>> request.get(uri, {
+			headers: {
+				'Accept': state.contentType
+			}
+		}).then(function(response) {
+			const result = state.responseDeserializer(String(response.data));
+			return ids.length === 1 ? [result] : result;
+		});
+	},
+
+	put(this: Storage<{}, {}>, items: {}[], options?: CrudOptions): Promise<UpdateResults<{}>> {
+		if (!items || items.length === 0) {
+			return Promise.reject(new Error('items can\'t be empty.'));
+		}
+
+		const state = instanceStateMap.get(this);
+		const ids = this.identify(items).filter((id) => !!id);
+		const isNewItems = ids.length < items.length;
+		let uri = state.target;
+
+		if (items.length === 1 && !isNewItems) {
+			uri += '/' + ids[0];
+		}
+
+		return <Promise<any>> request(uri, {
```

Comment:
This one is tricky. `requestSerializer` is used as the `object => string` serializer regardless of the type of the request. So it can't be used as an indicator of opt in custom request. I can probably add another option just for `put` custom request scenario, unless you have other ideas.